ouroboros-consensus/src/unstable-consensus-testlib/Test/Ouroboros/Consensus/ChainGenerator/Adversarial.hs

```diff
@@ -637,15 +637,29 @@ withinYS (Delta d) !mbYS !(RI.Race (C.SomeWindow Proxy win)) = case mbYS of
 --
 -- The count will be strictly smaller than the number of active slots in the given 'ChainSchema'.
 --
+-- The following precondition ensures that there are enough slots to produce an
+-- alternative chain schema with at least k+1 slots.
+--
+-- PRECONDITION: @schemaSize schedH >= s + k + d + 1@
```

Comment:
`genHonestRecipe` choses a length that is at least `2*s + d + 1`.

The assertion is a convoluted way to say that the intersection block must be chosen in the first `s-k` slots to make more probable that there will be $k + 1$ slots afterwards.

It might be too much of an assumption depending on how $Asc$ has been chosen. I didn't try to justify `k` over a smaller or higher value to accomplish that. It could be another value to adjust based on a chosen rate for false positives.

Reply:
Right, so $s + k + d + 1$ you chose because it will likely contain $k + 1$ active slots? That makes sense; any sequence of less than $2s-k + 1$ (maybe off by one) is technically not *guaranteed* to contain at least $k+1$ active slots just due to chain growth; but for the use case here, using an ad-hoc value seems fine, but I think it should be documented.